packages/blockchain-wallet-v4/src/walletCrypto/rng.js

```diff
@@ -0,0 +1,131 @@
+import randomBytes from 'randombytes'
+import assert from 'assert'
+import Either from 'data.either'
+import { isPositiveInteger, isHex } from '../utils/checks'
+
+const FORMAT = 'hex'
+const DEFAULT_BYTES = 32
+
+// Expose randomBytes for iOS to override
+export const _overrides = {
+  randomBytes
+}
+
+// xor :: Buffer -> Buffer -> Buffer
+export const xor = (a, b) => {
+  assert(
+    Buffer.isBuffer(a) && Buffer.isBuffer(b),
+    'Expected arguments to be buffers'
+  )
+
+  let length = Math.min(a.length, b.length)
+  let buffer = Buffer.alloc(length)
+
+  for (let i = 0; i < length; ++i) {
+    buffer[i] = a[i] ^ b[i]
+  }
+
+  return buffer
+}
+
+// getServerEntropy :: Int -> Promise (Either Buffer Error) Error
+export const getServerEntropy = (nBytes = DEFAULT_BYTES, api) => {
+  if (!isPositiveInteger(nBytes)) {
+    return Promise.resolve(Either.Left(
+      new Error('Must provide a positive integer to getServerEntropy')
+    ))
+  }
+
+  return api.getRandomBytes(nBytes, FORMAT).then(responseText => {
```

Comment:
Are non 200 error codes handled further upstream and result in an error that's caught and re-thrown in the catch below?

Reply:
A failed request will cause the promise returned by `createRng` to reject, so it's up to the caller to handle it. The code in this block only runs if the call randombytes was successful.

Reply:
OK, thanks!